normalize/diff.py

```diff
@@ -660,6 +665,19 @@ def compare_collection_iter(propval_a, propval_b, fs_a=None, fs_b=None,
             rev_key[(pk, seen[pk])] = k
             seen[pk] += 1
 
+    if options.recurse:
+        # we can be sure that both records have these keys
+        shared_keys = set(rev_key.values())
+        for key in shared_keys:
+            if (isinstance(propval_a, collections.Iterable) and
+               isinstance(propval_b, collections.Iterable)):
+
+                diffs = _diff_iter(propval_a[key], propval_b[key],
+                                   fs_a + [key], fs_b + [key], options)
+                for diff in diffs:
+                    yield diff
+                return
```

Comment:
is this at the right indent level? or should it be indented the same as `shared_keys`